### ISSUE TITLE
Usage of git tags

### ISSUE THREAD

**Issue by FlorianWilhelm** [CamDavidsonPilon/tdigest#29]
It would be really nice if tags were used in this project so that one could easily see if the current version 0.4.1.0 on PyPI actually refers to which commit and therefore also if a certain bugfix is included or not.

**Comment by CamDavidsonPilon** [comment]
I agree, though I find tag management to be not obvious to me. Can you provide your own workflow, or a workflow you have seen?

**Comment by FlorianWilhelm** [comment]
If you are using no tool for this purpose I think the default way is in order to make a new release is:
- Update the `version` to the new version number in setup.py, let's call the version X.Y.Z
- Also update `__version___` in `__init__.py` to X.Y.Z what most tools provide, check numpy, scipy etc.
  `import numpy as np; print(np.__version__)`
- Update your readme and changelog files to reflect the new version
- Run all your unit tests again and check that they work
- Commit this and call it "Version bump"
- Create a git tag X.Y.Z
- Build and submit the package to PyPI, i.e. `python setup.py bdist_wheel upload`

My personal recommendation is to just use [PyScaffold](https://pyscaffold.readthedocs.io/) which will provide an automised version scheme during development. When you want to release it's only:
- Update Readme and Changelog
- Commit everything
- Create a git tag
- Build and submit your package

PyScaffold will handle the rest like setting the version number everywhere.
